Tasks/AzureRmWebAppDeployment/msdeployutility.ts

```diff
@@ -139,32 +138,41 @@ export async function getMSDeployFullPath() {
     }
 }
 
-function getMSDeployVersion(registryKey: string): Q.Promise<String> {
-    var defer = Q.defer<String>();
-    regedit.list(registryKey)
-    .on('data', (entry) => {
-        var keys = entry.data.keys;
-        keys.sort();
-        if(parseFloat(keys[keys.length-1]) < 3) {
-            defer.reject(tl.loc("UnsupportedinstalledversionfoundforMSDeployversionshouldbealteast3orabove", keys[keys.length-1]));
-        }
-        defer.resolve(keys[keys.length-1]);
+function getMSDeployLatestRegKey(registryKey: string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    var regKey = new winreg({
+      hive: winreg.HKLM,
+      key:  registryKey
     })
-    .on('error', (error) => {
-        defer.reject(tl.loc("UnabletofindthelocationofMSDeployfromregistryonmachineError", error));
+
+    regKey.keys(function(err, subRegKeys) {
+        if(err) {
+            defer.reject(tl.loc("UnabletofindthelocationofMSDeployfromregistryonmachineError", err));
+        }
+        subRegKeys.sort();
```

Comment:
Is this sort safe to sort version number? This sort will be string sort while we need version numbers to be sorted right?